lib/mocha/mockery.rb

```diff
@@ -195,7 +195,7 @@ def logger
     private
 
     def method_exists_or_responds_to?(method, object)
-      object.method_exists?(method, true)
+      object.method_exists?(method, true) || object.respond_to?(method.to_sym)
```

Comment:
Can you explain how you determined that this is the only call site for `method_exists?` which needs the call to `respond_to?`?

Reply:
Because respond_to? was called only in the `if include_public_methods` conditional, and this is the only call site that sets `include_public_methods` to true.

Reply:
The only other call site is in the same file when we check for stubbing non-public methods. I found that by doing both a method reference search and a textual search.

Reply:
Ah, of course! That makes sense. Could you add that explanation to the commit note?

Reply:
Done